Source/WebCore/Modules/compression/DecompressionStreamDecoder.h

```diff
@@ -60,18 +72,36 @@ class DecompressionStreamDecoder : public RefCounted<DecompressionStreamDecoder>
     const size_t startingAllocationSize = 16384; // 16KB
     const size_t maxAllocationSize = 1073741824; // 1GB
     
-    bool initailized { false };
-    bool finish { false };
-    z_stream zstream;
+    bool m_initialized { false };
+    bool m_finish { false };
+    z_stream m_zstream;
+
+    bool m_usingAppleCompressionFramework { false };
+
+    inline ExceptionOr<RefPtr<JSC::ArrayBuffer>> decompress(const uint8_t* input, const size_t inputLength);
+
+#if PLATFORM(COCOA)
+    compression_stream m_stream;
+    ExceptionOr<RefPtr<JSC::ArrayBuffer>> decompressAppleCompressionFramework(const uint8_t* input, const size_t inputLength);
+    ExceptionOr<bool> initializeAppleCompressionFramework();
+#endif
 
     Formats::CompressionFormat m_format;
 
-    ExceptionOr<Vector<uint8_t>> decompress(const uint8_t* input, const size_t inputLength);
+    ExceptionOr<RefPtr<JSC::ArrayBuffer>> decompressZlib(const uint8_t* input, const size_t inputLength);
     ExceptionOr<bool> initialize();
 
     explicit DecompressionStreamDecoder(unsigned char format) 
         : m_format(static_cast<Formats::CompressionFormat>(format))
     {
+        std::memset(&m_zstream, 0, sizeof(m_zstream));
```

Comment:
Probably worth moving to cpp file

Reply:
Thanks. Moved both `memset`s to the initialize functions.